src/redux/sagas/utils/arbitraryTxs.ts

```diff
@@ -0,0 +1,80 @@
+import { Interface } from 'ethers/lib/utils';
+
+import { type AddTransactionTableModel } from '~v5/common/ActionSidebar/partials/forms/ArbitraryTxsForm/types.ts';
+
+/**
+ * Parses a string representation of an array into an array of values.
+ * It supports nested arrays and handles strings within arrays.
+ */
+const getArrayFromString = (value: string, type: string) => {
```

Comment:
This is loosely based on [Dapp implementation](https://github.com/JoinColony/colonyDapp/blob/c9ee24059989f5c285be700799242443986992fb/src/utils/safes/contractParserValidation.ts#L61) but has further improvements such as actually parsing nested arrays and handling potentially confusing strings containing commas and brackets (e.g. `["[", ","]`).